### PR TITLE
Extract logic from "dao apps" and "dao new"

### PR DESCRIPTION
**Pull request by macor161**
# 🦅 Pull Request

Extract logic from `dao apps` and `dao new` to the `lib` folder.
Supersedes #944

Part of the integration tests are dependent on IPFS. They will therefore be added once IPFS is included.

## 🚨 Test instructions

`dao new`
`dao apps`

### CODE REVIEW

**Comment by macor161** [comment]
hmm.. there is currently a weird Travis error on node 9 and 10. I'm looking into this.

File: packages/aragon-cli/src/commands/dao_cmds/new.js [hunk 1]
```diff
@@ -113,76 +84,35 @@ exports.task = async ({
...
-            templateInstance ||
-            new web3.eth.Contract(abi, ctx.repo.contractAddress)
-
-          const newInstanceTx = template.methods[fn](...fnArgs)
-          const estimatedGas = await newInstanceTx.estimateGas()
-          const { events } = await newInstanceTx.send({
-            from: ctx.accounts[0],
-            gas: await getRecommendedGasLimit(web3, estimatedGas),
+        task: async ctx => {
+          console.log('registry: ', apmOptions.ensRegistryAddress)
```
Comment (0x6431346e) [inline packages/aragon-cli/src/commands/dao_cmds/new.js]: :wave:

File: packages/aragon-cli/src/lib/dao/new.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+module.exports = async function({
+  repo,
+  web3,
+  templateInstance,
+  newInstanceMethod,
+  newInstanceArgs,
+  deployEvent,
+  gasPrice,
+}) {
+  const template =
```
Comment (0xGabi) [inline packages/aragon-cli/src/lib/dao/new.js]: I think this part is a bit awkward. Is been more and more clear that we need a clear distinction of how we handle the aragonPM information. Probably something to look a bit more in detail on current iteration after we merge this PR.

File: packages/aragon-cli/src/commands/dao_cmds/new.js (hunk 1 above)
Comment (0xGabi) [inline packages/aragon-cli/src/commands/dao_cmds/new.js]:
```suggestion
```

Comment (macor161) [reply packages/aragon-cli/src/commands/dao_cmds/new.js]: Good catch thanks! 😄

File: packages/aragon-cli/src/lib/dao/new.js (hunk 2 above)
Comment (macor161) [reply packages/aragon-cli/src/lib/dao/new.js]: I agree 💯
The `templateInstance` was originally used here https://github.com/aragon/aragon-cli/blob/develop/packages/aragon-cli/src/commands/run.js#L371 but we'll probably be able to simplify things once we start refactoring the `run` command.

File: packages/aragon-cli/src/commands/dao_cmds/new.js (hunk 1 above)
Comment (macor161) [reply packages/aragon-cli/src/commands/dao_cmds/new.js]: Thanks! :)

**Comment by dapplion** [comment]
According to AVA's documentation, most of the usage of `t.plan()` is not appropriate/necessary, see https://github.com/avajs/ava/blob/master/docs/recipes/when-to-use-plan.md

**Comment by macor161** [comment]
> According to AVA's documentation, most of the usage of t.plan() is not appropriate/necessary

@dapplion Ah I didn't know that thanks! It is fixed now.

**Comment by ajsantander** [comment]
@macor161 it seems that `dao new` is not working on this PR. It seems to hang on "Fetching template..."

**Comment by macor161** [comment]
> @macor161 it seems that dao new is not working on this PR. It seems to hang on "Fetching template..."

@ajsantander hmm strange it's working here but something is clearly wrong then. I will review this furthermore.

**Comment by macor161** [comment]
@0xGabi Thanks a lot!! 🙂

**Pull request merged**